workbench-core/environments/src/interfaces/updateEnvironmentTypeConfigsRequest.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { z } from 'zod';
+
+// eslint-disable-next-line @rushstack/typedef-var
+export const UpdateEnvironmentTypeConfigRequestParser = z
+  .object({
+    envTypeId: z.string(),
+    envTypeConfigId: z.string(),
+    params: z
```

Comment:
Same question - should we move params to make them props?

Reply:
for update I would prefer to separate the update props, that would save some validations, 
if I put all in the same level I would have to add undefined validations to avoid updating a prop that was not sent to the request, for create it makes sense to have it all in the same object

Reply:
I just found out I dont need the validations, the update method already validates undefined values, I will change it to be own properties

Reply:
moved props to top level